### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by piccaso** [50ButtonsEach/fliclib-linux-hci#35]
We would like to re(use) examples, libraries and binaries but I cant find any licensing information.
What licence applies to this components?

**Comment by Emill** [comment]
Hi. Sorry for that we haven't put any formal license up.

Basically, you may use examples and libraries in any way you wish as long as the purpose is to interact with our Flic buttons.

**Comment by piccaso** [comment]
Thanks for the fast reply!

does this also apply to the binaries?
Can we redistribute them with our project? 
(ie, create deb's for easy installation).

Right now i could not think of any other uses except to interact with Flic buttons, but i'll keep that in mind.

**Comment by Emill** [comment]
Yes you can redistribute them in a .deb, or any other way you like.

Well theoretically someone could make their own bluetooth buttons and create a tcp server implementing the same protocol...

**Comment by pfink** [comment]
@Emill: What you just did is putting this project under a rather nebulous 2-sentence-license by publicly giving this answer (which is meant to be nice, but still quite frustrating for those who need legal certainty). Why not take some minutes instead and choose an appropriate license?

http://choosealicense.com

Your backers deserve better!

**Comment by Emill** [comment]
We are currently in the process of adding proper licenses. We'll most likely add the same license here as the one in our Android sdk: https://github.com/50ButtonsEach/fliclib-android/commit/47fec8eb6cce351c1698abb2ebe8fb4bef44f71f

**Comment by pfink** [comment]
Good news, thanks for the update! :-)

**Comment by pfink** [comment]
Short remark regarding the linked license: If it would be applied 1:1 to this repo, if I understand it correctly, the redistribution as part of an open source licensed software would not be possible. Especially for projects like [OpenHab](https://github.com/openhab/openhab/wiki), this would be a pity.

Maybe it's possible to carve out the clientlibs and put them under a more permissive license, this would make it much easier for vendors of smart home integration software to use them and add Flic support to their products....

**Comment by Emill** [comment]
Thanks for your input. But can you explain why it would be impossible to redistribute this as part of an open source software? The license explicitly permits the sdk to be redistributed freely (under reasonable conditions).

**Comment by pfink** [comment]
Condition 4 says: _The Software may only be used for the purpose of interacting with products from Shortcut Labs AB._

I think, if a person would distributes a product containing Shortcut Labs code under, let's say, MIT License, this license does not contain this condition and cannot ensure that the user will not use the product (and therefore also the Shortcut Labs software as a part of it) for other purposes. So, Shortcut labs license and MIT are not compatible and you can't distribute a product containing Shortcut Labs software under MIT (or any other open source license) - this would, in my understaing, illegally compromise the Shortcut Labs license. Same applies to condition 3 (which is maybe at least compatible with Apache License 2.0 and Mozilla Public License 2.0 that contain a similar condition.)

I think it's a very similar situation to putting a library under GPL: This requires, that every software using the library, also use GPL or a GPL-compatible license. So, for example, putting a library under GPL, would probably also prevent it from being distributed with OpenHab which is distributed under the (incompatible) EPL 1.0. For that reason, licenses like [LGPL](https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License) has been made which contain specific remarks for software libraries that prevent such situations.

In my opinion, it's always a good approach for companies to protect the "heart" of their product with a little more restrictive license and to provide libraries that help to spread the product's support amongst other software so permissive as possible (e.g. using licenses like LGPL 3, MPL 2.0, AL 2.0, ...)

Disclaimer: I'm not a lawyer and this is no legal counseling. Just sharing my knowledge about free software licensing.

**Comment by Emill** [comment]
The "software" refers to the included files in the sdk, not the end product that might also contain software from other creators with other licenses. Our license does not force the developer to release the rest of a product under the same license (which GPL does). Does this make sense?

**Comment by pfink** [comment]
As I said, I'm not lawyer. But I still have serious doubts that this license text is covering what you're trying to achieve...

_The "software" refers to the included files in the sdk, not the end product that might also contain software from other creators with other licenses._

Condition 2 contains something similar, but it could be also understood as that you just not have to include the license text, but still have to comply with the license. Also, the point that the critical conditions 3 and 4 are following after 2, could imply that they always take effect, disregard whether it's a binary redistribution as part of an end product or not. In addition, there is no binary form of script languages like Python which are also part of this repo.

**Comment by wschlich** [comment]
flicd should REALLY be FOSS'ed! it's a shame that this is is only available as a precompiled binary blob.

**Comment by BandonRandon** [comment]
I'm not going to tell you how to license your code as that's something for you and your team to work out. What I will do is add my vote to adding a license. As others have pointed out there are number of benefits of adding a license officially to a project. 

I'd urge you to work quickly to find a license that both supports your mission as a company and legally protects the users of this library.

**Comment by pfink** [comment]
Hi @Emill @fabianbergmark,

I'd like to put a gentle reminder on this topic. Now I released my OpenHab Binding for Flic Button on the Eclipse Marketplace (https://marketplace.eclipse.org/content/flic-button-binding), but the current licensing situation is still not very clear why I did put some vague licensing hints on my release.

FOSSing at least the client libraries would really make it much more easy to develop and distribute Flic Button integrations based on flicd. At the moment as a developer I and also everyone using my Flic integration have to take some legal risks which is not quite nice... Here some examples about things that are currently quite unsure:

- compatibilty with other Open Source licenses very questionable. E.g. for GPL I'm quite sure that it's not compatible, so it would not be possible to distribute a release containing GPL and Shortcut Labs software in the same package
- redistribution conditions very unclear
- usage conditions of redistributions / modifications also unclear

Again I spent some hours to think about licensing risks rather then improving my Flic integration. And so again I'd like to vote for choosing a [popular, GPLv3-compatible license](https://www.dwheeler.com/essays/floss-license-slide.html) like MIT, BSD, Apache 2.0 or LGPL v3+ at least for the client libraries so that all people can develop & use Flic integrations without always having to worry about legal stuff.

I'm sorry that I intefer so hardly into a decision that is of course completely yours. But I think we share the same interests of making Flic buttons more awesome and I'd just like to help by making you aware how the licensing situation is making that much more hard and risky than it could be.
